yt_dlp/extractor/tvopengr.py

```diff
@@ -0,0 +1,150 @@
+
+
+class TVOpenGrWatchIE(TVOpenGrBaseIE):
+    IE_NAME = 'tvopengr:watch'
+    IE_DESC = 'tvopen.gr (and ethnos.gr) videos'
+    _VALID_URL = r'(?P<scheme>https?)://(?P<netloc>(?:www\.)?(?:tvopen|ethnos)\.gr)/watch/(?P<id>\d+)/(?P<slug>[^/]+)'
+    _API_ENDPOINT = '//www.tvopen.gr/templates/data/player'
+
+    _TESTS = [{
+        'url': 'https://www.ethnos.gr/watch/101009/nikoskaprabelosdenexoymekanenanasthenhsemethmethmetallaxhomikron',
+        'md5': '8728570e3a72e0f8d9475ba94859fdc1',
+        'info_dict': {
+            'id': '101009',
+            'title': 'md5:51f68773dcb6c70498cd326f45fefdf0',
+            'display_id': 'nikoskaprabelosdenexoymekanenanasthenhsemethmethmetallaxhomikron',
+            'description': 'md5:78fff49f18fb3effe41b070e5c7685d6',
+            'thumbnail': 'https://opentv-static.siliconweb.com/imgHandler/1920/d573ba71-ec5f-43c6-b4cb-d181f327d3a8.jpg',
+            'ext': 'mp4',
+            'upload_date': '20220109',
+            'timestamp': 1641686400,
+        },
+    }, {
+        'url': 'https://www.tvopen.gr/watch/100979/se28099agapaomenalla7cepeisodio267cmhthrargiapashskakias',
+        'md5': '38f98a1be0c577db4ea2d1b1c0770c48',
+        'info_dict': {
+            'id': '100979',
+            'title': 'md5:e021f3001e16088ee40fa79b20df305b',
+            'display_id': 'se28099agapaomenalla7cepeisodio267cmhthrargiapashskakias',
+            'description': 'md5:ba17db53954134eb8d625d199e2919fb',
+            'thumbnail': 'https://opentv-static.siliconweb.com/imgHandler/1920/9bb71cf1-21da-43a9-9d65-367950fde4e3.jpg',
+            'ext': 'mp4',
+            'upload_date': '20220108',
+            'timestamp': 1641600000,
+        },
+    }]
+
+    def _download_api_data(self, cid, scheme='https'):
+        url = f'{scheme}:{self._API_ENDPOINT}'
+        query = {'cid': cid}
+        return self._download_json(
+            url, query,
+            'Downloading JSON',
+            'Unable to download JSON',
+            query=query)
+
+    def _extract_formats_and_subs(self, options, video_id):
+        formats, subs = [], {}
+        for format_id, format_url in options.items():
+            if format_id not in ('stream', 'httpstream', 'mpegdash'):
+                continue
+            ext = determine_ext(format_url)
+            if ext == 'm3u8':
+                formats_, subs_ = self._extract_m3u8_formats_and_subtitles(
+                    format_url, video_id, 'mp4', m3u8_id=format_id,
+                    fatal=False)
+            elif ext == 'mpd':
+                formats_, subs_ = self._extract_mpd_formats_and_subtitles(
+                    format_url, video_id, 'mp4', fatal=False)
+            else:
+                formats.append({
+                    'url': format_url,
+                    'format_id': format_id,
+                })
+                continue
+            formats.extend(formats_)
+            subs.update(subs_)
```

Comment:
self._merge_subtitles

(just realized I overlooked this in glomex. I'll fix that)